### ISSUE TITLE
[LSP] Inconsistent hover tool tips.

### ISSUE THREAD

**Issue by linkpy** [godotengine/godot#43164]
**Godot version:**
3.2.3

**OS/device including version:**
Linux Mint

**Issue description:**

```
extends Node


var child: Node2D


func _ready():
	if get_child_count() > 0:
		child = get_child(0)
	
	var test := true
	print(test)
	
func _process(_dt: float) -> void:
	if child:
		var child_position: Vector2 = child.global_position
		var ang: float = atan2(child_position.y, 0)
		print(ang)
```

In VSCode :

- Hovering `test` in `print(test)` gives the correct tooltip.
- Hovering `child_position` in `atan2(child_position.y, 0)` gives no tooltip at all.
- Hovering `ang` in `print(ang)` gives no tooltip at all.

The issue is from the LSP server. VSCode correctly requests hover information to the LSP server, which replies with nothing.
Linked with https://github.com/godotengine/godot-vscode-plugin/issues/207.

**Steps to reproduce:**
See above

**Minimal reproduction project:**
/

**Comment by Razoric480** [comment]
This seems to be not a LSP issue, but a GDScript parsing issue. `ExtendGDScriptParser::search_symbol_defined_at_line()` receives the correct line number (line 17 for `ang`), but it does not include objects inside of the `if child:` block as a potential child to search through; that makes `_process()`'s only "child" `_dt`. So it never finds the `ang` symbol.

**Comment by akien-mga** [comment]
Fixed by #47338.
